iocore/utils/I_Machine.h

```diff
@@ -79,11 +81,15 @@ struct Machine {
                     );
   /// @return The global instance of this class.
   static self *instance();
+  bool is_self(const char *name);
+  void insert_id(const char *id);
```

Comment:
@zwoop thus far, only ParentSelection uses this when parsing the parent lists which might have ip address strings so, I think that I would continue add ip addresses to the string container..  What do you think of having also an is_self(ipSddrs&) function that would call inet_ntop() on the ipaddrs argument to create a temporary string to do the lookup on the string container?